src/main/java/core/basesyntax/OperationStrategyImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class OperationStrategyImpl extends OperationStrategy {
+    private Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public void execute(ShopStorage storage, FruitTransaction transaction) {
+        OperationHandler handler = operationHandlers.get(transaction.getOperation());
+        if (handler != null) {
+            handler.handle(storage, transaction);
+        } else {
+            System.out.println("Unknown operation: " + transaction.getOperation());
```

Comment:
Instead of using `System.out.println` for error handling, consider throwing an exception or using a logging framework to handle unknown operations. This will provide better error management and traceability in production environments.